### PR TITLE
Automatically normalize aliases for find* methods in VersionCatalog API

### LINKED ISSUE #18050: Document change in behaviour of VersionCatalog.findDependency for 7.2 in release notes
In `libs.versions.toml`: 
```
[versions]
kotlin = "1.5.21"

[libraries]
kotlin-stdlib-jdk8 = { module = "org.jetbrains.kotlin:kotlin-stdlib-jdk8", version.ref = "kotlin" }
```

In 7.1.1, this works in `buildSrc/src/main/kotlin/plugin.gradle.kts` 
```
val libs = extensions.getByType(VersionCatalogsExtension::class.java).named("libs")
dependencies {
    implementation(libs.findDependency("kotlin-stdlib-jdk8"))
}  
```
In 7.2, the above ☝️ now gives: 

```
   > Cannot convert the provided notation to an object of type Dependency: Optional.empty.
     The following types/formats are supported:
       - Instances of Dependency.
       .
       .
       .       
``` 

In 7.2, you have to do this instead in `buildSrc/src/main/kotlin/plugin.gradle.kts`, 
```
val libs = extensions.getByType(VersionCatalogsExtension::class.java).named("libs")
dependencies {
    implementation(libs.findDependency("kotlin.stdlib.jdk8"))
}  
```

Notice you have to swap `.` for `-` 

**Expected:**

Add this to the release notes for 7.2

### PR DESCRIPTION
**Pull request by asodja**
Fixes #18050

I figured to make it more user friendly, we can just automatically normalize aliases.

### CODE REVIEW

**Comment by asodja** [comment]
@gradle-bot help

**Comment by asodja** [comment]
@bot-gradle test QFL

**Comment by asodja** [comment]
@bot-gradle help

**Comment by asodja** [comment]
@bot-gradle test this

**Review by ljacomet — CHANGES_REQUESTED** [review]
The changes in here are 👍 for solving the non typed API issue. Though I would like to have the issue reference in the commit message itself.

However we should still clarify documentation around how declaring an alias like `my-plug` results in `libs.my.plug`.

In particular, the note in [that section](https://docs.gradle.org/current/userguide/platforms.html#sub:version-catalog-declaration) should become a section and be referenced from relevant places in the document.

Makes sense?

**Comment by asodja** [comment]
Makes sense, will check it.

**Comment by asodja** [comment]
New version for proof read:
[platforms.html.zip](https://github.com/gradle/gradle/files/7119769/platforms.html.zip)

**Comment by asodja** [comment]
New version for proof read:
[platforms.html.zip](https://github.com/gradle/gradle/files/7119801/platforms.html.zip)

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc [hunk 1]
```diff
@@ -74,17 +74,36 @@ We would generate the following type-safe accessors:
 - `libs.groovy.json`
 - `libs.androidx.awesome.lib`
 
-In case you want to avoid the generation of a subgroup accessor, we recommend to rely on case to differentiate.
+Where prefix `libs` comes from version catalog name.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc:77]: 💄 : Where the `libs` prefix comes from the version catalog name.

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc [hunk 2]
```diff
@@ -62,7 +62,7 @@ For example:
 - but `this.#is.not!`
 
 Then type safe accessors are generated for _each subgroup_.
-For example, given the following aliases:
+For example, given the following aliases in version catalog named `libs`:
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc:65]: 💄 : ... in a version catalog ...

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc [hunk 3]
```diff
@@ -228,6 +247,11 @@ Supported members of a version declaration are:
 
 Dependency declaration can either be declared as a simple string, in which case they are interpreted as `group:artifact:version` coordinates, or separating the version declaration from the group and name:
 
+[NOTE]
+====
+For aliases same rules apply as described in <<platforms.adoc#sub:mapping-aliases-to-accessors, aliases and their mapping to type safe accessors>> section.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc:252]: 💄 : For aliases, the rules described in the section <<platforms.adoc#sub:mapping-aliases-to-accessors, aliases and their mapping to type safe accessors>> apply as well.

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc [hunk 4]
```diff
@@ -261,6 +285,18 @@ or
 See the https://toml.io[TOML specification] for details.
 ====
 
+[[sub:programmatic-access-to-catalog]]
+=== Programmatic access to catalog
+
+You can also access version catalog programmatically with version catalog extension and version catalog API:
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc:291]: 💄 : ... with the version catalog extension and API:

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc [hunk 5]
```diff
@@ -261,6 +285,18 @@ or
...
+
+====
+include::sample[dir="snippets/dependencyManagement/catalogs-settings/groovy",files="build.gradle[tags=programmatic_access]"]
+include::sample[dir="snippets/dependencyManagement/catalogs-settings/kotlin",files="build.gradle.kts[tags=programmatic_access]"]
+====
+
+Check link:{javadocPath}/org/gradle/api/artifacts/VersionCatalog.html[version catalog API] for all supported methods.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc:298]: 💄 : Check the ...

**Review by ljacomet — COMMENTED** [review]
A couple suggestions on the added documentation:

**Review by ljacomet — APPROVED** [review]

**Comment by asodja** [comment]
@bot-gradle test QFL

**Review by ljacomet — APPROVED** [review]

**Comment by asodja** [comment]
@bot-gradle test and merge

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test and merge

**Pull request merged**